platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/EnhancedKeyStoreLoader.java

```diff
@@ -1165,4 +1175,340 @@ private interface AddressBookCallback {
+                                ERROR.getMarker(),
+                                "Failed to delete agreement key file {}",
+                                agreementKeyFile.getName());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Extracts the private keys and certificates from the PFX files and writes them to PEM files.
+     *
+     * @param pfxPrivateKeys  the map of extracted private keys.
+     * @param pfxCertificates the map of extracted certificates.
+     * @return the number of errors encountered during the extraction process.
+     * @throws KeyStoreException   if the underlying method calls throw this exception.
+     * @throws KeyLoadingException if the underlying method calls throw this exception.
+     */
+    private long extractPrivateKeysAndCertsFromPfxFiles(
+            final Map<NodeId, PrivateKey> pfxPrivateKeys, final Map<NodeId, Certificate> pfxCertificates)
+            throws KeyStoreException, KeyLoadingException {
+        final KeyStore legacyPublicStore = resolveLegacyPublicStore();
+        AtomicLong errorCount = new AtomicLong(0);
+
+        iterateAddressBook(addressBook, (i, nodeId, address, nodeAlias) -> {
+            if (isLocal(address)) {
+                // extract private keys for local nodes
+                final Path sPrivateKeyLocation = keyStoreDirectory.resolve("s-private-" + nodeAlias + ".pem");
+                final Path ksLocation = legacyPrivateKeyStore(nodeAlias);
+                if (!Files.exists(sPrivateKeyLocation) && Files.exists(ksLocation)) {
+                    logger.trace(
+                            STARTUP.getMarker(),
+                            "Extracting private signing key for node {} from file {}",
+                            nodeId,
+                            ksLocation.getFileName());
+                    final PrivateKey privateKey =
+                            readLegacyPrivateKey(nodeId, ksLocation, KeyCertPurpose.SIGNING.storeName(nodeAlias));
+                    pfxPrivateKeys.put(nodeId, privateKey);
+                    if (privateKey == null) {
+                        logger.error(
+                                ERROR.getMarker(),
+                                "Failed to extract private signing key for node {} from file {}",
+                                nodeId,
+                                ksLocation.getFileName());
+                        errorCount.incrementAndGet();
+                    } else {
+                        logger.trace(
+                                STARTUP.getMarker(),
+                                "Writing private signing key for node {} to PEM file {}",
+                                nodeId,
+                                sPrivateKeyLocation.getFileName());
+                        try {
+                            writePemFile(true, sPrivateKeyLocation, privateKey.getEncoded());
+                        } catch (IOException e) {
```

Comment:
`e` may be `final`

Reply:
found the IDE settings to turn this detection on.